streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -199,6 +201,7 @@ public <K, V> void send(final String topic,
                 log.trace("Failed record: (key {} value {} timestamp {}) topic=[{}] partition=[{}]", key, value, timestamp, topic, partition);
             }
         });
+        return recordSizeInBytes(keyBytes == null ? 0 : keyBytes.length, valBytes == null ? 0 : valBytes.length, topic, headers);
```

Comment:
I did it like this to avoid the extra/unnecessary null check for consumer records specifically, which already track the serialized size in bytes unlike the producer record. And unfortunately they don't inherit from a common interface/class -- but I added separate middle-man methods to handle them and moved the null check for the producer case there, should be addressed now